### DISCUSSION TITLE
Changing the SpanCategorizer threshold

### DISCUSSION THREAD

**Discussion by mbrunecky** [explosion/spaCy#8899] [Help: Coding & Implementations] [answered]
First, let me say THANK YOU for the SpanCategorizer, especially after applying #8855 (being able to get the individual scores).

SpanCategorizer is configured (config.cfg) using the 'threshold=0.5', which controls which spans are considered 'valid'. Depending upon project, one may probably tune it for the training.

However, when I load a trained SpanCategorizer model and use it for my production predictions, it uses the configuration from the training. But I would like to **lower** the threshold - so that I can 'see' even the 'not-so-good' predictions.

May be I am overlooking something, but IS THERE a WAY to **CHANGE** the threshold after the (trained) model is loaded?

**Comment by polm** [comment, marked as answer]
The config file is saved with the serialized model as `config.cfg`. You should just be able to modify the threshold in the config file before loading. Does that not work for you?

If you want to do it after the model is loaded, the config is stored as a dict on the pipeline component and you can just modify it. Something like:

```
import spacy
nlp = spacy.load(...)
spancat = nlp.get_pipe("spancat")
spancat.cfg["threshold"] = 0.2
```

**Reply by mbrunecky** [reply]
I do not like editing the 'model' under the hood, because it is hard to 'productize' - somebody somehow would have to make that happen. Not to mention that _model/config.cfg_ and _model/spancat/cfg_ are two separate files and it is not clear which one to edit.
I prefer changing the loaded configuration programmatically. I was just not aware of the pipe.cfg attribute. 
Thank you!
